python_tests/test_python_frontend.py

```diff
@@ -1468,6 +1469,97 @@ def fusion_func(fd: FusionDefinition) :
             self.assertEqual(torch.real(inputs[0]), nvf_out[0])
             self.assertEqual(torch.imag(inputs[0]), nvf_out[1])
 
+    def test_pad(self):
+        inputs = [
+            torch.testing.make_tensor((2, 3), dtype=torch.float32, device="cuda"),
+        ]
+
+        def fusion_func(fd: FusionDefinition):
+            t0 = fd.from_pytorch(inputs[0])
+
+            t1 = fd.ops.pad(t0, [1, 1, 1, 1])
+            fd.add_output(t1)
+
+            # tensor method version
+            t2 = t0.pad([1, 1, 1, 1])
+            fd.add_output(t2)
+
+            # zero padding in some dims
+            t3 = fd.ops.pad(t0, [0, 0, 2, 3])
+            fd.add_output(t3)
+
+            # zero padding in all dims
+            t4 = fd.ops.pad(t0, [0, 0, 0, 0])
+            fd.add_output(t4)
+
+            # no padding provided in first dim
+            t5 = fd.ops.pad(t0, [2, 3])
+            fd.add_output(t5)
+
+            # test padding with a value other than 0
+            fill_val = fd.define_constant(2.0)
+            t6 = fd.ops.pad(t0, [2, 3], fill_val)
+            fd.add_output(t6)
+
+            # test padding with a value other than 0 with tensor method
+            t7 = t0.pad([2, 3], fill_val)
+            fd.add_output(t7)
+
+        nvf_out, _ = self.exec_nvfuser(fusion_func, inputs)
+
+        self.assertEqual(F.pad(inputs[0], [1, 1, 1, 1]), nvf_out[0])
+        self.assertEqual(F.pad(inputs[0], [1, 1, 1, 1]), nvf_out[1])
+        self.assertEqual(F.pad(inputs[0], [0, 0, 2, 3]), nvf_out[2])
+        self.assertEqual(F.pad(inputs[0], [0, 0, 0, 0]), nvf_out[3])
+        self.assertEqual(F.pad(inputs[0], [2, 3]), nvf_out[4])
+        self.assertEqual(F.pad(inputs[0], [2, 3], "constant", 2.0), nvf_out[5])
+        self.assertEqual(F.pad(inputs[0], [2, 3], "constant", 2.0), nvf_out[6])
+
+    def test_pad_cache(self):
```

Comment:
@jjsjann123 I _think_ this tests everything we need to here. The fill value is an optional scalar, so `None` is different from the default value of `0`. I guess we'll get a new kernel if an explicit 0 is passed instead of `None`.

Reply:
LGTM. Thanks for adding tests.